tests/Unit/Evolution/Systems/CurvedScalarWave/Worldtube/SingletonActions/Test_ChangeSlabSize.cpp

```diff
@@ -0,0 +1,213 @@
+                runner, 0) == time_step_id);
+        CHECK(ActionTesting::get_databox_tag<worldtube_chare,
+                                             ::Tags::Next<::Tags::TimeStepId>>(
+                  runner, 0) == next_time_step_id);
+        CHECK(ActionTesting::get_databox_tag<worldtube_chare, ::Tags::TimeStep>(
+                  runner, 0) == step);
+        CHECK(ActionTesting::get_databox_tag<worldtube_chare,
+                                             ::Tags::Next<::Tags::TimeStep>>(
+                  runner, 0) == next_step);
+        CHECK(ActionTesting::get_databox_tag<worldtube_chare, ::Tags::Time>(
+                  runner, 0) == time);
+      };
+  using inbox_variables_type =
+      Variables<tmpl::list<CurvedScalarWave::Tags::Psi,
+                           ::Tags::dt<CurvedScalarWave::Tags::Psi>>>;
+  {
+    ActionTesting::MockRuntimeSystem<MockMetavariables<Dim>> runner{{}};
+    create_new_runner(make_not_null(&runner));
+    auto& worldtube_inbox =
+        ActionTesting::get_inbox_tag<worldtube_chare,
+                                     Tags::SphericalHarmonicsInbox<Dim>>(
+            make_not_null(&runner), 0);
+    worldtube_inbox[time_step_id_1][element_ids.at(0)] = inbox_variables_type{};
+    // ChangeSlabSize but only 1/2 elements have sent
+    CHECK_FALSE(ActionTesting::next_action_if_ready<worldtube_chare>(
+        make_not_null(&runner), 0));
+
+    // send from other element
+    worldtube_inbox[time_step_id_1][element_ids.at(1)] = inbox_variables_type{};
+    CHECK(ActionTesting::next_action_if_ready<worldtube_chare>(
+        make_not_null(&runner), 0));
+    // same slab size so nothing shoud have changed.
+    check_time_tags(runner, time_step_id_1,
+                    time_stepper->next_time_id(time_step_id_1, step_1), step_1,
+                    step_1, start_time_1.value());
+#ifdef SPECTRE_DEBUG
+    const TimeStepId time_step_id_new_slab_number(true, 2, start_time_1);
+    // sending a second time step id, should fail
+    worldtube_inbox[time_step_id_new_slab_number][element_ids.at(0)] =
+        inbox_variables_type{};
+    CHECK_THROWS_WITH(
+        ActionTesting::next_action<worldtube_chare>(make_not_null(&runner), 0),
+        Catch::Contains("Received data from two different time step ids."));
+#endif
+  }
+
+  {
+    ActionTesting::MockRuntimeSystem<MockMetavariables<Dim>> runner{{}};
+    create_new_runner(make_not_null(&runner));
+    auto& worldtube_inbox =
+        ActionTesting::get_inbox_tag<worldtube_chare,
+                                     Tags::SphericalHarmonicsInbox<Dim>>(
+            make_not_null(&runner), 0);
+    // different slab start -> invalid
+    const double slab_2_start = 1.7;
+    const Slab slab_2(slab_2_start, slab_1_end);
+    const auto start_time_2 = slab_2.start();
+    const TimeStepId time_step_id_2(true, 1, start_time_2);
+    worldtube_inbox[time_step_id_2][element_ids.at(0)] = inbox_variables_type{};
+#ifdef SPECTRE_DEBUG
```

Comment:
Move this up to contain the entire block.  Since all the checks are in in the `SPECTRE_DEBUG` section there's no reason to do the setup in release mode.